src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/ClientCreds/JsonWebToken.cs

```diff
@@ -128,7 +130,13 @@ private string Encode(IClientAssertionCertificate credential)
         {
             // Header segment
             string jsonHeader = EncodeHeaderToJson(credential);
+            string emptyX5cClaim = "\"x5c\":\"\",";
 
+            //Remove x5c claim if no certificate is available
+            if (jsonHeader.Contains(emptyX5cClaim))
+            {
+                jsonHeader = jsonHeader.Remove(jsonHeader.IndexOf(emptyX5cClaim), emptyX5cClaim.Length);
```

Comment:
this looks quite hacky and brittle. Think about leveraging ifdef to light up the x5c claim in the header class